### PR TITLE
Rework AdSense detection logic and update mock data.

### PR DESCRIPTION
**Pull request by asvinb**
## Summary

Addresses issue:

- #5052 

## PR Author Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/modules/adsense/components/setup/v2/SetupAccount.js [hunk 1]
```diff
@@ -109,9 +109,9 @@ export default function SetupAccount( { account } ) {
 	}
 
 	if (
-		state === SITE_STATE_NEEDS_ATTENTION ||
-		state === SITE_STATE_REQUIRES_REVIEW ||
-		state === SITE_STATE_GETTING_READY
+		accountState === STATE_NEEDS_ATTENTION ||
+		site?.state === STATE_REQUIRES_REVIEW ||
+		site?.state === STATE_GETTING_READY
```
Comment (tofumatt) [inline assets/js/modules/adsense/components/setup/v2/SetupAccount.js:114]: These look like new constants specific to account state: https://github.com/google/site-kit-wp/pull/5109/files#diff-7a4be23cb854366543b75e18911a77d95f74ca436570d142fff54c6b20d8e429R29-R31

But they're used for site state. Why create the new constants?

**Review by tofumatt — COMMENTED** [review]
Looks good, just a question about the new constants.

File: assets/js/modules/adsense/components/setup/v2/SetupAccount.js [hunk 2]
```diff
@@ -78,22 +78,22 @@ export default function SetupAccount( { account } ) {
...
 			setAccountStatus( ACCOUNT_STATUS_NO_CLIENT );
-		} else if ( state === SITE_STATE_NEEDS_ATTENTION ) {
+		} else if ( accountState === STATE_NEEDS_ATTENTION ) {
 			setAccountStatus( ACCOUNT_STATUS_NEEDS_ATTENTION );
-		} else if ( state === SITE_STATE_REQUIRES_REVIEW ) {
+		} else if ( site?.state === STATE_REQUIRES_REVIEW ) {
 			setAccountStatus( ACCOUNT_STATUS_CLIENT_REQUIRES_REVIEW );
-		} else if ( state === SITE_STATE_GETTING_READY ) {
+		} else if ( site?.state === STATE_GETTING_READY ) {
 			setAccountStatus( ACCOUNT_STATUS_CLIENT_GETTING_READY );
```
Comment (felixarntz) [inline assets/js/modules/adsense/components/setup/v2/SetupAccount.js:92]: This is still not correct. Please review https://github.com/google/site-kit-wp/issues/5052#issuecomment-1101907230 and https://github.com/google/site-kit-wp/pull/5083#discussion_r852525037.

You're now looking at the _site's_ state. But as mentioned in my previous comments, you need to look at the _client's_ state. The client is different from the account and also different from the site. We need to look at the AFC client in the account and use _that_ state.

There is no dedicated store selector to get that client yet, we only have `getClients( accountID )`. So we may want to add something like `getAFCClient( accountID )` that calls `getClients` and parses out the AFC client (or returns `null` if there is none).

File: assets/js/modules/adsense/components/setup/v2/SetupAccount.js (hunk 1 above)
Comment (felixarntz) [reply assets/js/modules/adsense/components/setup/v2/SetupAccount.js:114]: See my above comment, this is incorrect looking at the `site`, it needs to look at the client for the last two states here.

File: assets/js/modules/adsense/datastore/constants.js [hunk 3]
```diff
@@ -25,3 +25,7 @@ export const SITE_STATE_NEEDS_ATTENTION = 'NEEDS_ATTENTION';
 export const SITE_STATE_REQUIRES_REVIEW = 'REQUIRES_REVIEW';
 export const SITE_STATE_GETTING_READY = 'GETTING_READY';
 export const SITE_STATE_READY = 'READY';
```
Comment (felixarntz) [inline assets/js/modules/adsense/datastore/constants.js:27]: I would advise we remove these four constants, as they can become quite confusing in combination with our own states. To clarify, the states here are the possible API values, which can apply to the `state` field of accounts, clients, and sites. So _if_ we want to have constants for them, let's use the `STATE_` prefix like below. Or for even more clarity, we could use `API_STATE_`, just to clarify these are not our own account or site status values.

**Review by felixarntz — COMMENTED** [review]
This is still not correct, see detailed comments.

Another nit-pick: There's a typo throughout this component, it's _AFC_ client, not _ACF_ client :)

**Pull request merged**